app/helpers/trades_helper.rb

```diff
@@ -0,0 +1,5 @@
+module TradesHelper
+  def user_can_offer_trade?
+    logged_in? && current_user.id != @trade.user_id && @trade.status == Trade::STATUSES['open'] && current_user.console == @user.console
```

Comment:
Wow this is quite a check, great job pulling it into its own easy to read method.

Like I mentioned in another comment some of these might also be worth pulling into their own methods, but obviously you could go overboard on that, which wouldn't help the code either.  Just pointing it out as an area to evaluate, not necessarily something that needs to be done.  Part of it is preference